src/Internal/ConcurrentIteratorChannel.php

```diff
@@ -71,24 +73,42 @@ public function onClose(\Closure $onClose): void
 
     public function receive(?Cancellation $cancellation = null): mixed
     {
-        if (!$this->receive->continue($cancellation)) {
-            $this->close();
-            throw new ChannelException("The channel closed while waiting to receive the next value");
+        try {
+            if ($this->receive->continue($cancellation)) {
+                return $this->receive->getValue();
+            }
+        } catch (DisposedException $exception) {
+            // Throw exception below
         }
 
-        return $this->receive->getValue();
+        throw new ChannelException(
+            "The channel closed while waiting to receive the next value",
+            0,
+            $exception ?? null,
+        );
     }
 
     public function send(mixed $data): void
     {
-        if ($data === null) {
-            throw new ChannelException("Cannot send null on a channel");
-        }
```

Comment:
Why did you remove this restriction now?

Reply:
I guess this was missed when we allowed null on channels.